Update the template used for generating _get_type() functions so that they use
g_once_init_{enter,leave}_pointer where appropriate.

See the upstream issue:
  https://gitlab.gnome.org/GNOME/glib/-/issues/2842
for more information.

Upstream-Status: Inappropriate [CHERI specific]

--- a/gdk-pixbuf/gdk-pixbuf-enum-types.c.template	2022-10-25 17:53:28.000000000 +0100
+++ b/gdk-pixbuf/gdk-pixbuf-enum-types.c.template	2025-02-03 19:12:57.672540878 +0000
@@ -13,9 +13,9 @@
 GType
 @enum_name@_get_type (void)
 {
-    static gsize g_define_type;
+    static GType g_define_type;
 
-    if (g_once_init_enter (&g_define_type)) {
+    if (g_once_init_enter_pointer (&g_define_type)) {
         static const G@Type@Value values[] = {
 /*** END value-header ***/
 
@@ -28,7 +28,7 @@
         };
         GType enum_type = g_@type@_register_static (g_intern_static_string ("@EnumName@"), values);
 
-        g_once_init_leave (&g_define_type, enum_type);
+        g_once_init_leave_pointer (&g_define_type, enum_type);
     }
     return g_define_type;
 }
